src/transformers/models/llama/configuration_llama.py

```diff
@@ -141,6 +141,16 @@ class LlamaConfig(PretrainedConfig):
 
     model_type = "llama"
     keys_to_ignore_at_inference = ["past_key_values"]
+    # Default tensor parallel plan for base model `LlamaModel`
+    _base_model_tp_plan = {
```

Comment:
Do we want to allow this for external use by removing `_` so that we can allow users to define tp plan tweaks from `config.json`? 

Given that, shall we as well allow for providing custom tp plan as input to `LlamaConfig()` that overrides the default?

Reply:
Yeah, good idea. We can make this public once we prove things work.

Reply:
Yep, `base_model_tp_plan` should be supported as input to the PreTrainedConfig!

Reply:
So, this variable `base_model_tp_plan` has to be added to PreTrainedConfig https://github.com/huggingface/transformers/blob/a769ed45e17c44fd17b85c025863c4e4f2f73634/src/transformers/configuration_utils.py#L50 with a default value as an empty dict `{}` which I believe is best possible default for any config sub class inheriting.

Reply:
Thanks @kmehant @ArthurZucker for the suggestion. I moved `base_model_tp_plan` to `PretrainedConfig` in the latest commit.